lib/module.js

```diff
@@ -3,8 +3,8 @@ require('./core');
 const isGenerator = require('./isGenerator');
 const defaultBaseModule = {};
 
-module.exports = function (baseModule, moduleDefinition) {
-  if (arguments.length === 1) {
+module.exports = function (baseModule, moduleDefinition, doNotWrapInGenRunCallback) {
```

Comment:
@mrlotfi I reviewed everything, consider making this changes. Would you please write some tests for the third argument.